gitpod-io/gitpod
# Dedicated Setup - SSO Setup

## Description

Adds the SSO Setup step to the dedicated setup flow. Configuration can be provided, and then verified. This should launch a login flow. Upon completion the user should now be logged in as themselves (not the admin user), and see the complete view.

| Header | Header |
|--------|--------|
| ![image](https://user-images.githubusercontent.com/367275/235789590-655ee28b-9cf3-4f3e-9126-4ae69db309ae.png) | ![image](https://user-images.githubusercontent.com/367275/235789807-522daef2-c243-41f0-8416-30e74493db9f.png) | 

The flow is still only launched behind the `dedicated-setup=force` query param. This will get integrated with a proper check against the state of the cell in a followup.

## Related Issue(s)

Fixes [WEB-245](https://linear.app/gitpod/issue/WEB-245)

## How to test

TBD - need to hook up to a proper check on if the cell needs the setup flow launched to make this easier to test.

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - bmh-dedica0d3380689d</li>
	<li><b>🔗 URL</b> - <a href="https://bmh-dedica0d3380689d.preview.gitpod-dev.com/workspaces" target="_blank">bmh-dedica0d3380689d.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options:

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

</details>

<details>
<summary>Installer Options</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold
